src/System.Management.Automation/engine/CommandBase.cs

```diff
@@ -266,6 +266,406 @@ internal void InternalDispose(bool isDisposing)
+            /// </summary>
+            public string LightGray { get; set; } = "\x1b[97m";
+
+            /// <summary>
+            /// Gets or sets the color light green.
+            /// </summary>
+            public string LightGreen { get; set; } = "\x1b[92m";
+
+            /// <summary>
+            /// Gets or sets the color light magenta.
+            /// </summary>
+            public string LightMagenta { get; set; } = "\x1b[95m";
+
+            /// <summary>
+            /// Gets or sets the color light red.
+            /// </summary>
+            public string LightRed { get; set; } = "\x1b[91m";
+
+            /// <summary>
+            /// Gets or sets the color light yellow.
+            /// </summary>
+            public string LightYellow { get; set; } = "\x1b[93m";
+
+            /// <summary>
+            /// Gets or sets the color magenta.
+            /// </summary>
+            public string Magenta { get; set; } = "\x1b[35m";
+
+            /// <summary>
+            /// Gets or sets the color read.
+            /// </summary>
+            public string Red { get; set; } = "\x1b[31m";
+
+            /// <summary>
+            /// Gets or sets the color white.
+            /// </summary>
+            public string White { get; set; } = "\x1b[37m";
+
+            /// <summary>
+            /// Gets or sets the color yellow.
+            /// </summary>
+            public string Yellow { get; set; } = "\x1b[33m";
+
+            /// <summary>
+            /// Set as RGB (Red, Green, Blue).
+            /// </summary>
+            /// <param name="red">Byte value representing red.</param>
+            /// <param name="green">Byte value representing green.</param>
+            /// <param name="blue">Byte value representing blue.</param>
+            /// <returns>String representing ANSI code for RGB value.</returns>
+            public string FromRgb(byte red, byte green, byte blue)
+            {
+                return $"\x1b[38;2;{red};{green};{blue}m";
+            }
+
+            /// <summary>
+            /// The color set as RGB as a single number.
+            /// </summary>
+            /// <param name="rgb">RGB value specified as an integer.</param>
+            /// <returns>String representing ANSI code for RGB value.</returns>
+            public string FromRgb(int rgb)
+            {
+                byte red, green, blue;
+                blue = (byte)(rgb & 0xFF);
+                rgb >>= 8;
+                green = (byte)(rgb & 0xFF);
+                rgb >>= 8;
+                red = (byte)(rgb & 0xFF);
+
+                return FromRgb(red, green, blue);
+            }
+        }
+
+        /// <summary>
+        /// Contains background colors.
+        /// </summary>
+        public class BackgroundColor
+        {
+            /// <summary>
+            /// Gets or sets the color black.
+            /// </summary>
+            public string Black { get; set; } = "\x1b[40m";
```

Comment:
I think this kind of goes into a philosophical discussion on whether the shell should own color definitions or should that only be the terminal.  I'm sort of leaning towards it being the domain of the terminal, in which case it would make sense these are only getters.  I suppose it would be better to have them as getters only as it's not breaking to make them settable in the future.